src/pydata_sphinx_theme/toctree.py

```diff
@@ -101,8 +101,18 @@ def unique_html_id(base_id: str):
         return next(get_or_create_id_generator(base_id))
 
     @cache
-    def generate_header_nav_before_dropdown(n_links_before_dropdown):
-        """The cacheable part."""
+    def generate_header_nav_before_dropdown(
+        n_links_before_dropdown,
+    ) -> Tuple[str, List[str]]:
+        """Return html for navbar and dropdown.
+
+        Given the number of links before the dropdown, return the html for the navbar,
+        as well as the list of links to but in a dropdown.
+
+        Returns:
+            - HTML str for the navbar
```

Comment:
I think it would be a simple refactor that might help make this function easier to understand if it returns 

- list of HTML str for navbar
- list of HTML str for the dropdown

Reply:
I think better than that, the cached function should not return html but the data necessary to build the html. But that require some refactor. 

I'll make an issue to suggest doing a refactor for later.